radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/ConstraintMachine.java

```diff
@@ -139,7 +139,7 @@ public ReadableAddrs immutableIndex() {
 						.filter(SystemParticle.class::isInstance)
 						.findFirst()
 						.or(() -> store.loadAddr(txn, addr))
-						.or(() -> Optional.of(new SystemParticle(0, 0, 0))); // A bit of a hack
+						.or(() -> Optional.of(new SystemParticle(0, 0, 0, null))); // A bit of a hack
```

Comment:
The whole supplier can be replaced with method reference and returned value - into constant. This allows to avoid unnecessary allocation and whole code would be a bit cleaner and readable.

Reply:
Not worth making it cleaner now as this needs to be refactored towards a better architecture overall.